File: app/src/main/java/br/com/popularmovies/moviedetail/trailers/adapters/TrailerAdapter.java
package br.com.popularmovies.moviedetail.trailers.adapters;

import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import java.util.List;

import br.com.popularmovies.R;
import br.com.popularmovies.services.movieService.response.MovieTrailer;

public class TrailerAdapter extends RecyclerView.Adapter<TrailerAdapter.ReviewViewHolder> {

    private List<MovieTrailer> mTrailers;

    public TrailerAdapter(List<MovieTrailer> trailers) {
        this.mTrailers = trailers;
    }

    @NonNull
    @Override
    public TrailerAdapter.ReviewViewHolder onCreateViewHolder(@NonNull ViewGroup viewGroup, int i) {
        Context context = viewGroup.getContext();
        View view = LayoutInflater.from(context).inflate(R.layout.movie_trailer, viewGroup, false);
        return new TrailerAdapter.ReviewViewHolder(view);
    }

    @Override
    public void onBindViewHolder(@NonNull TrailerAdapter.ReviewViewHolder movieViewHolder, int position) {
        MovieTrailer movieTrailer = mTrailers.get(position);
        movieViewHolder.mTitle.setText(("Author: ").concat(movieTrailer.getName()));

    }

    @Override
    public int getItemCount() {
        return mTrailers.size();
    }

    class ReviewViewHolder extends RecyclerView.ViewHolder {
        private ImageView mMediaPlay;
        private TextView mTitle;
        private ImageView mShare;

        ReviewViewHolder(@NonNull View itemView) {
            super(itemView);
            mMediaPlay = itemView.findViewById(R.id.iv_trailer_play);
            mTitle = itemView.findViewById(R.id.tv_trailer_title);
            mShare = itemView.findViewById(R.id.iv_trailer_share);
        }
    }
}

File: app/src/main/java/br/com/popularmovies/moviedetail/trailers/adapters/TrailerAdapter.kt
package br.com.popularmovies.moviedetail.trailers.adapters

import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ImageView
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView
import br.com.popularmovies.R
import br.com.popularmovies.entities.movie.MovieTrailer
import br.com.popularmovies.moviedetail.trailers.Constants
import br.com.popularmovies.moviedetail.trailers.adapters.TrailerAdapter.TrailerViewHolder

class TrailerAdapter(private val movieTrailers: List<MovieTrailer>,
                     private val mOnTrailerClickListener: TrailerClickListener) :
        RecyclerView.Adapter<TrailerViewHolder>() {
    override fun onCreateViewHolder(viewGroup: ViewGroup, i: Int): TrailerViewHolder {
        val context = viewGroup.context
        val view = LayoutInflater.from(context).inflate(R.layout.movie_trailer, viewGroup, false)
        return TrailerViewHolder(view)
    }

    override fun onBindViewHolder(movieViewHolder: TrailerViewHolder, position: Int) {
        val movieTrailer = movieTrailers[position]
        movieViewHolder.mTitle.text = movieTrailer.name
        movieViewHolder.mMediaPlay.setOnClickListener { mOnTrailerClickListener.onPlay(movieTrailer.key) }
        movieViewHolder.mShare.setOnClickListener { mOnTrailerClickListener.onShare(Constants.YOUTUBE_URL + movieTrailer.key) }
    }

    override fun getItemCount(): Int {
        return movieTrailers.size
    }

    inner class TrailerViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
        val mMediaPlay: ImageView = itemView.findViewById(R.id.iv_trailer_play)
        val mTitle: TextView = itemView.findViewById(R.id.tv_trailer_title)
        var mShare: ImageView = itemView.findViewById(R.id.iv_trailer_share)
    }
}